src/main/clojure/clara/macros.clj

```diff
@@ -196,10 +196,11 @@
 (sc/defn ^:always-validate compile-alpha-nodes
   [alpha-nodes :- [schema/AlphaNode]]
   (vec
-   (for [{:keys [condition beta-children env]} alpha-nodes
+   (for [{:keys [id condition beta-children env]} alpha-nodes
          :let [{:keys [type constraints fact-binding args]} condition]]
 
-     {:type (com/effective-type type)
+     {:id id
+      :type (com/effective-type type)
       :alpha-fn (com/compile-condition type (first args) constraints fact-binding env)
       :children (vec beta-children)})))
```

Comment:
It seems odd to me that we build a map here and then use it to construct alpha nodes later in assemble-session while building the beta nodes directly in this namespace, but changing that is probably out of scope here.  Maybe there's a reason for this that I'm not remembering offhand.

Reply:
I agree that is a little weird the only thing i think that is forcing this is, the access to the compiled beta nodes. At this point the beta-children are just ids of the nodes that will be constructed later.